src/server/auth.ts

```diff
@@ -43,6 +44,46 @@ passport.use(
 	)
 );
 
+// create hacker from user
+export const createHackerFromUser = async (user: InstanceType<User>, done: Function) => {
+	const createdHacker = await HackerModel.create({
+		email: user.email,
+		status: Status.Created,
+		user: user._id,
+	});
+	if (createdHacker) {
+		logger.debug(user);
+		done(null, user);
+	} else {
+		done(null, false);
+	}
+};
+
+// create a new user
+export const createAuthenticatedUser = async (
+	authType: AuthType,
+	email: string,
+	id: any,
+	done: Function
+) => {
+	logger.debug('> Creating user.....');
+	const newUser = {
+		authLevel: AuthLevel.HACKER,
+		authType,
+		email,
+		githubId: authType === AuthType.GITHUB ? id : undefined,
+		googleId: authType === AuthType.GOOGLE ? id : undefined,
+		password: `${authType}!123`,
```

Comment:
Oooooooooooook unless we don't expose the username/password surface on production builds (which seems annoying), we absolutely 100% cannot have a default password that's insecure (and has its generation code exposed to the world!). We either need to not require a password (better) or create (with a crypto designed RNG) and hash one for each user.

Create a high priority issue for this if you don't want to get it done now but as it stands, this is a massive vulnerability.

Reply:
this is copied from before, will create issue